src/service/subscription-service.ts

```diff
@@ -53,23 +53,37 @@ export class SubscriptionService {
     grantId: string,
   ): Promise<Response> {
     const endpoint: string = `${
-      SubscriptionService.endpoint.emailParam + encodeURIComponent(emailAddress)
+      SubscriptionService.endpoint.userParam + encodeURIComponent(emailAddress)
     }/${SubscriptionService.endpoint.grantIdParam + grantId}`;
     const result = await SubscriptionService.client.get(endpoint);
     return result.data;
   }
 
-  async deleteSubscriptionByEmailAndGrantId(
-    emailAddress: string,
+  async getSubscriptionBySubAndGrantId(
+    sub: string,
     grantId: string,
     unsubscribeReference?: string,
-  ) {
-    const endpoint = `${
-      SubscriptionService.endpoint.emailParam + encodeURIComponent(emailAddress)
+  ): Promise<Response> {
+    const endpoint: string = `${
```

Comment:
string type here is inferred
```suggestion
    const endpoint = `${
```